lib/deadpool-cyclone/src/instance/cyclone/local_uds.rs

```diff
@@ -486,6 +476,243 @@ fn temp_path_and_socket_from(
+                }
+                if spec.action {
+                    cmd.arg("--enable-action-run");
+                }
+
+                Ok(LocalInstanceRuntime {
+                    runtime_strategy: LocalUdsRuntimeStrategy::LocalProcess,
+                    command: Some(cmd),
+                    child: None,
+                    container_id: None,
+                    docker_api: None,
+                })
+            }
+            LocalUdsRuntimeStrategy::LocalDocker => {
+                let mut cmd = vec![
+                    String::from("--bind-uds"),
+                    sock.to_string_lossy().to_string(),
+                    String::from("--decryption-key"),
+                    String::from("/tmp/key"),
+                    String::from("--lang-server"),
+                    String::from("/tmp/langserver"),
+                    String::from("--enable-watch"),
+                ];
+                if let Some(limit_requests) = spec.limit_requests {
+                    cmd.push(String::from("--limit-requests"));
+                    cmd.push(limit_requests.to_string())
+                }
+                if let Some(timeout) = spec.watch_timeout {
+                    cmd.push(String::from("--watch-timeout"));
+                    cmd.push(timeout.as_secs().to_string());
+                }
+                if spec.ping {
+                    cmd.push(String::from("--enable-ping"));
+                }
+                if spec.resolver {
+                    cmd.push(String::from("--enable-resolver"));
+                }
+                if spec.action {
+                    cmd.push(String::from("--enable-action-run"));
+                }
+
+                let docker = Docker::connect_with_local_defaults()?;
+                let rand_string: String = thread_rng()
+                    .sample_iter(&Alphanumeric)
+                    .take(10)
+                    .map(char::from)
+                    .collect();
+
+                let socket_dir = sock
+                    .parent()
+                    .expect("socket path not available")
+                    .display()
+                    .to_string();
+                let mounts = vec![
+                    Mount {
+                        source: Some(socket_dir.clone()),
+                        target: Some(socket_dir.clone()),
+                        typ: Some(MountTypeEnum::BIND),
+                        ..Default::default()
+                    },
+                    Mount {
+                        source: Some(spec.cyclone_decryption_key_path),
+                        target: Some(String::from("/tmp/key")),
+                        typ: Some(MountTypeEnum::BIND),
+                        ..Default::default()
+                    },
+                    Mount {
+                        source: Some(
+                            spec.lang_server_cmd_path
+                                .as_path()
+                                .to_string_lossy()
```

Comment:
Similarly to the above, we may want to use `to_str`. This is because [`to_string_lossy`](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy) does the following:

> Any non-Unicode sequences are replaced with [U+FFFD REPLACEMENT CHARACTER](https://doc.rust-lang.org/std/char/constant.REPLACEMENT_CHARACTER.html).

I don't think we want to eat that kind of error despite it being impossible to hit.